dci-agent-setup/roles/setup_dci_agent/tasks/main.yaml

```diff
@@ -0,0 +1,78 @@
+---
+- name: Ensure required variables are defined
+  fail: msg="{{ item }} value missing and is required"
+  when: "{{ item }} is not defined"
+  with_items:
+    - dci_group_name
+    - remotcis
+
+- name: Ensure that the role is being used on a supported OS
+  assert:
+    that: (ansible_os_family == 'RedHat' and ansible_distribution_major_version == '7') or
+          (ansible_distribution == 'Ubuntu')
+
+# I think RHEL has different steps for installing EPEL
+- name: install EPEL
+  yum:
+    name: epel-release
+    state: present
+  when: ansible_distribution == "CentOS"
+
+- name: install pip
+  package:
+    name: python-pip
+    state: present
+
+- name: install a couple of neccesary python packages
+  pip:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - junit_xml   # allows usage of the junit callback plugin
+    - xunitmerge  # allows user to merge junit into one file
+
+- name: create group for DCI files
+  group:
+    name: "{{ dci_group_name }}"
+    state: present
+
+- name: create directory for storing api keys
+  file:
+    path: /etc/dci
+    state: directory
+    mode: 0770
+    owner: root
+    group: "{{ dci_group_name }}"
+
+- name: set up connection information
+  template:
+    src: env.sh
+    dest: /etc/dci/{{ item.name }}.sh
+    owner: root
+    group: "{{ dci_group_name }}"
+    mode: 0770
+  with_items: "{{remotcis}}"
+  no_log: true
+
+- name: red hat
```

Comment:
The two blocks are unecessary, they can be merged in one block (for now) until more platform are supported. Even then only the key for differentiation would change but logic would stay the same.

```
block:
  - debug: msg="Running tasks for {{ ansible_os_family }} systems"
  - include: "{{ ansible_os_family }}.yaml"
```

If at a latter point we want to rely on `ansible_distribution` then we'll switch the key